cmds/exp/rush/builtins.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2012 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// builtins: cd, exit, pwd, echo
+// extras: rushinfo
+
+package main
+
+import (
+	"runtime"
+	"errors"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+)
+
+// Error messages
+var errCdUsage = errors.New("usage: cd <directory-to-change-to>")
+var errExitUsage = errors.New("usage: exit <|0-255|>")
+
+// Add builtins to the shell
+func init() {
```

Comment:
This is actually not really the best way to do this IMHO. If you look at the old cd.go, you can see that its init function did what you are doing here. Nobody cares about rush any more, so I'm not going to worry about it, but I actually think this is taking things the wrong direction.